### PR TITLE
MODORDERS-222 - Order Line has inactive Access Provider

### PR DESCRIPTION
**Pull request by khandramai**
[MODORDERS-222](https://issues.folio.org/browse/MODORDERS-222) - Order Line has inactive Access Provider

## Purpose
PO Line numbers should be displayed in the error modal if associated access provider is inactive/not found.

## Approach
Included poLineNumber population to error response during Vendor/AccessProvider validation.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/VendorHelper.java [hunk 1]
```diff
@@ -59,15 +62,15 @@ public VendorHelper(Map<String, String> okapiHeaders, Context ctx, String lang)
         .thenApply(vendor -> {
           VendorStatus status = VendorStatus.valueOf(vendor.getVendorStatus().toUpperCase());
           if(status != VendorStatus.ACTIVE) {
-            errors.add(createErrorWithId(ORDER_VENDOR_IS_INACTIVE, id));
+            errors.add(createErrorWithId(ORDER_VENDOR_IS_INACTIVE, id, poLines));
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/VendorHelper.java]: vendor is at the PO level, I'm not sure it makes sense to add an error for every single POL

Comment (khandramai) [reply src/main/java/org/folio/rest/impl/VendorHelper.java]: @craigmcnally, done. Thanks!

File: src/main/java/org/folio/rest/impl/VendorHelper.java [hunk 2]
```diff
@@ -150,6 +153,24 @@ private Error createErrorWithId(ErrorCodes errorCodes, String id) {
...
+   *
+   * @param id vendor's/access provider's identifier
+   * @param errorCodes error code
+   * @param poLines list of composite PoLines
+   * @return {@link Error} with id of failed vendor/access provider
+   */
+  private Error createErrorWithId(ErrorCodes errorCodes, String id, List<CompositePoLine> poLines) {
+    Error error = createErrorWithId(errorCodes, id);
+    for (CompositePoLine poLine : poLines) {
+      if (poLine.getPoLineNumber() != null) {
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/VendorHelper.java]: This logic does not take into account if the PO Line's access provider id equals to vendor `id` but just adds all PO Line numbers as parameters.
I verified this by `PurchaseOrdersApiTest#testPutOrdersByIdToChangeStatusToOpenInactiveVendor` unit test adding in the beginning of the test
```java
    for (int i = 0; i < reqData.getCompositePoLines().size(); i++) {
      reqData.getCompositePoLines().get(i).setPoLineNumber("number-" + i);
    }
```
After updates above the test fails when there is one active and one inactive access provider specified

File: src/main/java/org/folio/rest/impl/VendorHelper.java [hunk 3]
```diff
@@ -91,7 +92,12 @@ public VendorHelper(Map<String, String> okapiHeaders, Context ctx, String lang)
   public CompletableFuture<Errors> validateAccessProviders(CompositePurchaseOrder compPO) {
     CompletableFuture<Errors> future = new VertxCompletableFuture<>(ctx);
     Set<String> ids = verifyAndGetAccessProvidersIdsList(compPO);
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/VendorHelper.java]: Looks like this set is no longer needed because this is the same as `poLinesMap.keySet()`

**Pull request merged**